tests/target-browser/target_browser_test.go

```diff
@@ -266,405 +250,664 @@ var _ = Describe("Target Browser Tests", func() {
+			var (
+				backupPlanUIDs          []string
+				noOfBackupPlansToCreate = 4
+				noOfBackupsToCreate     = 1
+				once                    sync.Once
+				isLast                  bool
+				backupUID               string
+			)
+			BeforeEach(func() {
+				once.Do(func() {
+					backupUID = guid.New().String()
+					createTarget(true)
+					output, _ := exec.Command(createBackupScript, strconv.Itoa(noOfBackupPlansToCreate),
+						strconv.Itoa(noOfBackupsToCreate), "true", backupUID).Output()
+					log.Info("Shell Script Output: ", string(output))
+					backupPlanUIDs = verifyBackupPlansAndBackupsOnNFS(noOfBackupPlansToCreate, noOfBackupPlansToCreate*noOfBackupsToCreate)
+					time.Sleep(2 * time.Minute) // wait to sync data on target browser
+				})
+			})
 
-		It("Should filter backupPlans on TVK Instance UID and order by backupPlan name ascending order", func() {
-			isLast = true
-			for _, value := range tvkInstanceIDValues {
-				args := []string{cmdGet, cmdBackupPlan, flagTvkInstanceUIDFlag, value, flagOrderBy, Name}
-				backupPlanData := runCmdBackupPlan(args)
-				Expect(len(backupPlanData)).To(Equal(1))
-				Expect(backupPlanData[0].InstanceID).Should(Equal(value))
-				for index := 0; index < len(backupPlanData)-1; index++ {
-					Expect(backupPlanData[index].BackupPlanName <= backupPlanData[index+1].BackupPlanName).Should(BeTrue())
+			AfterEach(func() {
+				if isLast {
+					deleteTarget(true)
+					for _, backupPlans := range backupPlanUIDs {
+						_, err := shell.RmRf(filepath.Join(TargetLocation, backupPlans))
+						Expect(err).To(BeNil())
+					}
 				}
-			}
+			})
+
+			It(fmt.Sprintf("Should fail if flag %s is given without value", cmd.BackupStatusFlag), func() {
+				args := []string{cmdGet, cmdBackup, flagBackupPlanUIDFlag, backupPlanUIDs[1]}
+				args = append(args, commonArgs...)
+				args = append(args, flagBackupStatus)
+				command := exec.Command(targetBrowserBinaryFilePath, args...)
+				output, err := command.CombinedOutput()
+				Expect(err).Should(HaveOccurred())
+				Expect(string(output)).Should(ContainSubstring(fmt.Sprintf("flag needs an argument: %s", flagBackupStatus)))
+			})
+			It(fmt.Sprintf("Should get one backup for specific backup UID %s", backupUID), func() {
+				args := []string{cmdGet, cmdBackup, backupUID}
+				backupData := runCmdBackup(args)
+				Expect(len(backupData)).To(Equal(1))
+				Expect(backupData[0].UID).To(Equal(backupUID))
+			})
+
+			It(fmt.Sprintf("Should get two backup for backup UID %s", backupUID), func() {
+				args := []string{cmdGet, cmdBackup, backupUID, backupUID}
+				backupData := runCmdBackup(args)
+				Expect(len(backupData)).To(Equal(2))
+				Expect(backupData[0].UID).To(Equal(backupUID))
+			})
+
+			It("Should get zero backup using backupPlanUID 'invalid'", func() {
+				args := []string{cmdGet, cmdBackup, flagBackupPlanUIDFlag, "invalid"}
+				backupData := runCmdBackup(args)
+				Expect(len(backupData)).Should(Equal(0))
```

Comment:
why there's no error 404? check once

Reply:
We are not getting 404. Getting status code 200

Reply:
should get 404 if we are making req. `/backupplan/:uid` and uid is not present.. update this to @niranjan-gavade or @vishutrilio